staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -163,7 +153,43 @@ func (g *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID str
 
 // InstanceMetadataByProviderID returns metadata of the specified instance.
 func (g *Cloud) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
-	return nil, fmt.Errorf("unimplemented")
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
+	defer cancel()
+
+	md := &cloudprovider.InstanceMetadata{}
```

Comment:
nit: instead of defining it here, I think the cleaner flow would be:
```
...
nodeAddresses, err := nodeAddressesFromInstance(instance); err != nil {
  return nil, err
}
return &cloudprovider.InstanceMetadata{
  ProviderID: providerID,
  Type: lastComponent(instanceMachineType),
  NodeAddresses: nodeAddresses,
}, nil
```

Reply:
addressed in all implementations, except for azure(makes multiple return cleaner)